File: gdx2d-desktop/src/hevs/gdx2d/demos/scrolling/objects/Cloud.java
package hevs.gdx2d.demos.scrolling.objects;

import hevs.gdx2d.components.bitmaps.BitmapImage;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.interfaces.DrawableObject;


/**
 * A simple cloud that does nothing special
 * @author Pierre-Andre Mudry (mui)
 */
public class Cloud implements DrawableObject {

	final BitmapImage cloudImage = new BitmapImage("data/images/cloud_1.png");
	public int x, y;
	
	public Cloud(int x, int y){
		this.x = x;
		this.y = y;
	}
	
	@Override
	public void draw(GdxGraphics g) {
		g.drawPicture(x, y, cloudImage);
	}

}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/scrolling/objects/Cloud.scala
package ch.hevs.gdx2d.demos.scrolling.objects

import ch.hevs.gdx2d.components.bitmaps.BitmapImage
import ch.hevs.gdx2d.lib.GdxGraphics
import ch.hevs.gdx2d.lib.interfaces.DrawableObject

/** A simple cloud that does nothing special. */
class Cloud(var x: Int, var y: Int) extends DrawableObject {
  private val cloudImage = new BitmapImage("images/cloud_1.png")

  override def draw(g: GdxGraphics): Unit =
    g.drawPicture(x.toFloat, y.toFloat, cloudImage)
}
